backend/modules/parsers/unstructured_io.py

```diff
@@ -36,11 +36,11 @@ class UnstructuredIoParser(BaseParser):
         ".xlsx",
     ]
 
-    def __init__(self, max_chunk_size: int = 2000, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
```

Comment:
```suggestion
    def __init__(self, *, max_chunk_size: int = 2000, **kwargs):
```

Reply:
Ideally we should create change the base class of Parsers to pydantic BaseModel

In the future we want to iterate over all registered parsers, get their openapi / json schema so that we can show full config on UI

Only possible challenge I see is how we declare private fields with pydantic
Maybe Pydantic v2 has better solution for this.

---

Another alternative can be

```
class BaseParser(abc.ABC):
    class InitArgs(BaseModel):
        pass
    def __init__(self, args: InitArgs):
        ...


class UnstructuredIoParser(BaseParser):
    class InitKwargs(BaseParser.InitArgs):
        max_chunk_size: int = 2000
    def __init__(self, args: InitArgs):
        ...
```

I don't like the extra toil in unwrapping `args` but seems like a decent tradeoff

> class names can be changed

Reply:
https://docs.pydantic.dev/latest/concepts/models/#private-model-attributes

v2 does have private attrs, just need to make the field name start with `_`

Reply:
Keeping this open for future reference